attendance/views.py

```diff
@@ -119,4 +131,62 @@ def get_item(dictionary, key):
 
 
 def get_date_from_template_returned_string(string_date):
-    return datetime.strptime(string_date, "%B %d, %Y").date()
+    for date_format in ["%B %d, %Y", "%b. %d, %Y", "%bt. %d, %Y"]:
+        try:
+            return datetime.strptime(string_date, date_format).date()
+        except ValueError:
+            pass
+    raise ValueError("time data {!r} does not match any expected date format".format(string_date))
+
+
+def update_course_attendance_statistic(course_id):
+    class_attendance = Attendance.objects.filter(
+        classroom_id=course_id, date__range=["2000-01-01", datetime.today().date()]
+    )
+    average = get_average_attendance_from_list(class_attendance)
+    classroom = Classroom.objects.get(id=course_id)
+    classroom.attendance_summary = {"attendance_statistic": round(average * 100, 2)}
+    classroom.save()
+
+
+def get_course_attendance_statistic(course_id):
+    return Classroom.objects.get(id=course_id).attendance_summary.get(
+        "attendance_statistic"
+    )
+
+
+def create_absence_notifications(request, student_list, absences, date):
```

Comment:
I made a note about the callsite for this, but the way student_list and absences are used together here is error-prone as there's no guarantee that they are in the same order. Would be better to take only the absences list with the assumption that the students are already loaded in that query set.

Reply:
Resolved in 989e97ca4de16df31e508a8eeb5154ade44b3bd5